code/modules/reagents/chemistry_properties/prop_negative.dm

```diff
@@ -41,26 +41,127 @@
 /datum/chem_property/negative/toxic/process_overdose(mob/living/M, var/potency = 1, delta_time)
 	M.apply_damage(potency * delta_time, TOX)
 
-/datum/chem_property/negative/toxic/process_critical(mob/living/M, var/potency = 1, delta_time)
-	M.apply_damage(2 * potency * delta_time, TOX)
+/datum/chem_property/negative/toxic/process_critical(mob/living/M, var/potency = 1)
+	M.apply_damage(potency * POTENCY_MULTIPLIER_VHIGH, TOX)
+
+/datum/chem_property/negative/toxic/reaction_obj(var/obj/O, var/volume, var/potency = 1)
+	if(istype(O,/obj/effect/alien/weeds/))
+		var/obj/effect/alien/weeds/alien_weeds = O
+		alien_weeds.take_damage(25 * potency) // Kills alien weeds on touch
+		return
+	if(istype(O,/obj/effect/glowshroom))
+		qdel(O)
+		return
+	if(istype(O,/obj/effect/plantsegment))
+		if(prob(50)) qdel(O)
+		return
+	if(istype(O,/obj/structure/machinery/portable_atmospherics/hydroponics))
```

Comment:
what is this istype madness 💀

Reply:
code that mainly was transferred from what we already have, see code/modules/reagents/chemistry_reagents/toxin.dm 166-193
the only change is that its now applied to a property instead of a specific chemical